clang/lib/Sema/SemaChecking.cpp

```diff
@@ -5267,6 +5267,15 @@ bool CheckAllArgsHaveFloatRepresentation(Sema *S, CallExpr *TheCall) {
   return false;
 }
 
+void SetInputRepresentationReturnType(Sema *S, CallExpr *TheCall,
```

Comment:
Consider renaming the function to something like `SetElementTypeAsReturnType` - to hopefully convey its functionality a bit more specifically.

Reply:
I'll rename it but need language to indicate that the provided return type is  just the base type ie if the arguments are vectors and not scalars then what we are setting is just the vector element type.